### PR TITLE
Add github actions

### PR DESCRIPTION
**Pull request by rullzer**
Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

**Comment by rullzer** [comment]
@skjnldsv @ChristophWurst do I need the whole eslint import thing? this is not done autmatically by the nextcloud/eslint-plugin package?

**Comment by ChristophWurst** [comment]
Which import?

**Comment by rullzer** [comment]
> Which import?

https://github.com/nextcloud/bruteforcesettings/pull/176/checks?check_run_id=497345531 fails.
But I only included the nextcloud eslint pacakge. Not eslint itself. I assumed this would be pulled in automagically.

**Comment by ChristophWurst** [comment]
Ah no. Because it's [just a peer](https://github.com/nextcloud/eslint-config/blob/c4619485bef3cafdb8ebdd65f2853d50d5a28b30/package.json#L13). And they don't get installed automatically. You have to install it manually.

**Comment by skjnldsv** [comment]
> Ah no. Because it's [just a peer](https://github.com/nextcloud/eslint-config/blob/c4619485bef3cafdb8ebdd65f2853d50d5a28b30/package.json#L13). And they don't get installed automatically. You have to install it manually.

Yep, we can't force importing configs :)
We can only warn

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+	extends: [
+		'nextcloud'
```
Comment (skjnldsv) [inline .eslintrc.js]:
```suggestion
		'@nextcloud'
```

File: package.json [hunk 2]
```diff
@@ -20,12 +20,27 @@
...
-    "css-loader": "^3.4.2",
     "nextcloud-axios": "^0.2.1",
     "vue": "^2.6.11"
   },
   "devDependencies": {
+    "@nextcloud/eslint-plugin": "^1.1.0",
+    "babel-eslint": "^10.0.3",
+    "css-loader": "^3.4.2",
+    "eslint": "^5.16.0",
+    "eslint-config-nextcloud": "^0.1.1",
```
Comment (skjnldsv) [inline package.json]:
```suggestion
    "@nextcloud/eslint-config": "^1.1.0",
```

File: package.json [hunk 3]
```diff
@@ -20,12 +20,27 @@
...
+    "eslint-config-standard": "^12.0.0",
+    "eslint-loader": "^3.0.3",
+    "eslint-plugin-import": "^2.20.1",
+    "eslint-plugin-nextcloud": "^0.3.0",
```
Comment (skjnldsv) [inline package.json]:
```suggestion
```

**Comment by skjnldsv** [comment]
Now update the lockfile ;)
(you can cheat with compilebot :see_no_evil: )

**Comment by rullzer** [comment]
@skjnldsv any idea how to solve https://github.com/nextcloud/bruteforcesettings/pull/176/checks?check_run_id=497471166 the lint errors?

**Comment by skjnldsv** [comment]
> @skjnldsv any idea how to solve [#176 (checks)](https://github.com/nextcloud/bruteforcesettings/pull/176/checks?check_run_id=497471166) the lint errors?

Temporary exception of the line or don't use oc_default :)

**Comment by skjnldsv** [comment]
> oc_defaults is deprecated: use OC.theme instead, this will be removed in Nextcloud 20

**Comment by skjnldsv** [comment]
It's actually not used, you can remove https://github.com/nextcloud/bruteforcesettings/search?q=oc_defaults&unscoped_q=oc_defaults

**Pull request merged**